### DISCUSSION TITLE
WebGPU

### DISCUSSION THREAD

**Discussion by kittendevv** [konsoletyper/teavm#728] [Ideas]
> Dear TeaVM dev team
Could you guys ad web GPU support. Thanks CodingKitten

**Comment by konsoletyper** [comment]
@CodingKiten you can do it yourself

**Reply by kittendevv** [reply]
No lmao. I can only code C# HTML,Css and Js

**Reply by konsoletyper** [reply]
So why are you asking to improve Java library?

**Reply by kittendevv** [reply]
Cuz it whould make everything compiled with TeaVM like 10X faster

**Reply by konsoletyper** [reply]
To make everything work faster, you need first to use WebGPU in your project (which is eaglercraft in your case). So without any Java skills how are you doing this? You should ask whoever maintains eaglercraft to add WebGPU support on their side.

**Reply by kittendevv** [reply]
In the eaglercraft subreddit, they said teaVM needs to support WebGPU first

**Reply by konsoletyper** [reply]
They are lying

**Reply by kittendevv** [reply]
So TeaVM supporst WebGPU?

**Reply by konsoletyper** [reply]
TeaVM does not need to support WebGPU. TeaVM is a compiler, not a **(insert your JS API)** compatibility layer. They must write WebGPU support themselves.

**Reply by kittendevv** [reply]
Ok thanks for the conservation

**Comment by masondaniels** [comment]
I am working on (and plan to maintain) a library that will make it fairly simple to use WebGPU with TeaVM. It's called WebGPUJ4W. Please take a look at the repository [here](https://github.com/masondaniels/WebGPUJ4W). It is in the very early stages so right now it only supports vertex and fragment shaders and has minimal features (compute shaders [along with multiple compute shader support] should be added in the following days).
